test/specs/elements/Label/Label-test.js

```diff
@@ -0,0 +1,272 @@
+        .should.have.tagName('div')
+    })
+
+    it('has no detail when not defined', () => {
+      shallow(<Label />)
+        .should.not.have.descendants('.detail')
+    })
+
+    it('adds a detail last child', () => {
+      shallow(<Label detail={faker.hacker.noun()}><br /></Label>)
+        .children()
+        .last()
+        .should.match('.detail')
+    })
+
+    it('adds the value as the detail text', () => {
+      const detail = faker.hacker.noun()
+      shallow(<Label detail={detail} />)
+        .find('.detail')
+        .should.contain.text(detail)
+    })
+  })
+
+  describe('detailLink', () => {
+    it('has no detail when not defined', () => {
+      shallow(<Label />)
+        .should.not.have.descendants('.detail')
+    })
+
+    it('renders detail as an a tag', () => {
+      shallow(<Label detailLink={faker.hacker.noun()} />)
+        .find('.detail')
+        .should.have.tagName('a')
+    })
+
+    it('adds a detail last child', () => {
+      shallow(<Label detailLink={faker.hacker.noun()}><br /></Label>)
+        .children()
+        .last()
+        .should.match('.detail')
+    })
+
+    it('adds the value as the detail text', () => {
+      const detailLink = faker.hacker.noun()
+      shallow(<Label detailLink={detailLink} />)
+        .find('.detail')
+        .should.contain.text(detailLink)
+    })
+  })
+
+  describe('icon', () => {
+    it('has no i when not defined', () => {
+      shallow(<Label />)
+        .should.not.have.descendants('i')
+    })
+
+    it('adds a i as first child', () => {
+      shallow(<Label icon={faker.hacker.noun()}><br /></Label>)
+        .childAt(0)
+        .should.match('i')
+    })
+
+    it('adds the value to the Icon className', () => {
+      const className = faker.hacker.noun()
+      shallow(<Label icon={className}><br /></Label>)
+        .find('Icon')
+        .should.have.className(className)
+    })
+  })
+
+  describe('image', () => {
+    it('has no img when prop is not defined', () => {
+      shallow(<Label />)
+        .should.not.have.descendants('img')
+    })
+
+    it('adds image className when defined', () => {
+      const src = faker.internet.avatar()
+      shallow(<Label image={src} />)
+        .should.have.className('image')
+    })
+
+    it('adds image className when there is an img child', () => {
+      shallow(<Label><img /></Label>)
+        .should.have.className('image')
+    })
+
+    it('adds image className when there is an Image child', () => {
+      shallow(<Label><Image /></Label>)
+        .should.have.className('image')
+    })
+
+    it('adds an img as first child', () => {
+      shallow(<Label image={faker.internet.avatar()}><br /></Label>)
+        .find('Image')
+        .should.have.tagName('img')
+    })
+
+    it('adds the value to the Image className', () => {
+      const src = faker.internet.avatar()
+      shallow(<Label image={src} />)
+        .find('Image')
+        .should.have.prop('src', src)
+    })
```

Comment:
Note, we're also using [`chai-enzyme`](https://github.com/producthunt/chai-enzyme) to give us these end of chain assertions like `should.have.prop()`.

If we had only `enzyme` we'd have awkward chai assertions made on enzyme values.